src/AppInstallerCLICore/Commands/PinCommand.cpp

```diff
@@ -0,0 +1,245 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+#include "pch.h"
+#include "PinCommand.h"
+#include "Workflows/CompletionFlow.h"
+#include "Workflows/PinFlow.h"
+#include "Workflows/WorkflowBase.h"
+#include "Resources.h"
+
+namespace AppInstaller::CLI
+{
+    using namespace AppInstaller::CLI::Execution;
+    using namespace AppInstaller::Utility::literals;
+    using namespace std::string_view_literals;
+
+    // TODO: Create this link with the docs!
+    static constexpr std::string_view s_PinCommand_HelpLink = "https://aka.ms/winget-command-pin"sv;
+
+    std::vector<std::unique_ptr<Command>> PinCommand::GetCommands() const
+    {
+        return InitializeFromMoveOnly<std::vector<std::unique_ptr<Command>>>({
+            std::make_unique<PinAddCommand>(FullName()),
+            std::make_unique<PinRemoveCommand>(FullName()),
+            std::make_unique<PinListCommand>(FullName()),
+            std::make_unique<PinResetCommand>(FullName()),
+        });
+    }
+
+    Resource::LocString PinCommand::ShortDescription() const
+    {
+        return { Resource::String::PinCommandShortDescription };
+    }
+
+    Resource::LocString PinCommand::LongDescription() const
+    {
+        return { Resource::String::PinCommandLongDescription };
+    }
+
+    std::string PinCommand::HelpLink() const
+    {
+        return std::string{ s_PinCommand_HelpLink };
+    }
+
+    void PinCommand::ExecuteInternal(Execution::Context& context) const
+    {
+        OutputHelp(context.Reporter);
+    }
+
+    std::vector<Argument> PinAddCommand::GetArguments() const
+    {
+        return {
+            Argument::ForType(Args::Type::Query),
+            Argument::ForType(Args::Type::Id),
+            Argument::ForType(Args::Type::Name),
+            Argument::ForType(Args::Type::Moniker),
+            Argument::ForType(Args::Type::Tag),
+            Argument::ForType(Args::Type::Command),
+            Argument::ForType(Args::Type::Exact),
+            Argument{ "version"_liv, 'v', Args::Type::GatedVersion, Resource::String::GatedVersionArgumentDescription, ArgumentType::Standard },
+            Argument::ForType(Args::Type::Source),
+            Argument::ForType(Args::Type::CustomHeader),
+            Argument::ForType(Args::Type::AcceptSourceAgreements),
+            Argument::ForType(Args::Type::Force),

```

Comment:
Curious what is the scenario for --force in pin add?

Reply:
The scenario I was thinking of is overwriting the pin. Like `winget pin add Pkg` adds a pin, then `winget pin add Pkg --version 1.*` would fail because it exists, but if you add `--force`, it overwrites it. But honestly, I just added it because it was in the spec and I was hoping you had a scenario in mind :D

Reply:
That makes sense. I guess I thought about it some time ago and now my memory does not serve me well.

Reply:
Even more curious - should `--force` be added to the list of generic arguments (like `--wait` and `--verbose-logs`)? In the worst case, it doesn't change the behavior, and in most cases it will remove a few lines of code.